File: complex_ws/src/complex/scripts/complex_pub.py
#!/usr/bin/env python3

import rospy
from complex.msg import comp
import random


def com_pub():
    rospy.init_node('pub_complex', anonymous=True)
    pub = rospy.Publisher("complex", comp, queue_size=1)
    inp_complex = comp()
    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        inp_complex.Real = round(random.random() * 50,1)
        inp_complex.Imaginary = round(random.random() * 50,1)

        pub.publish(inp_complex)
        rate.sleep()

if __name__=="__main__":
    try:
        com_pub()
    except rospy.ROSInterruptException:
        pass
    
    
    

File: complex_ws/src/complex/src/complex_pub.cpp
#include <ros/ros.h>
#include <complex/comp.h>

int main(int argc, char**argv)
{
    ros::init(argc,argv, "pub_complex");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<complex::comp>("complex", 1000);
    ros::Rate rate(1);

    complex::comp inp_complex;

    while (ros::ok())
    {
        inp_complex.Real=1+(rand() %50);
        inp_complex.Imaginary=1+(random() %50);
        pub.publish(inp_complex);
        rate.sleep();
    }

    return 0;
}

